src/service/gif.js

```diff
@@ -1,9 +1,9 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(search?.trim() !== '' ? `/chat/gif/search?q=${search.trim()}` : '/chat/gif/featured')
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are generally good, but there's a potential issue with the way the `search` parameter is being used. If `search` contains any characters that aren't valid in a URL, this could cause problems. To avoid this, you should use `encodeURIComponent` to ensure that `search` is properly encoded before it's appended to the URL.

```diff
- .get(search?.trim() !== '' ? `/chat/gif/search?q=${search.trim()}` : '/chat/gif/featured')
+ .get(search?.trim() !== '' ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}` : '/chat/gif/featured')
```
This will ensure that any special characters in the search string are correctly encoded and won't cause issues with the request URL.


<!-- This is an auto-generated comment by OSS CodeRabbit -->